cktap/proto.py

```diff
@@ -234,8 +234,13 @@ def _set_derivation(self, path: List[int], cvc):
         if not all_hardened(path):
             raise ValueError("All path components must be hardened")
 
-        _, resp = self.send_auth('derive', cvc, path=path, nonce=pick_nonce())
+        picked_nonce = pick_nonce()
+        _, resp = self.send_auth('derive', cvc, path=path, nonce=picked_nonce)
+        msg = b'OPENDIME' + self.card_nonce + picked_nonce + resp["chain_code"]
+        assert len(msg) == 8 + CARD_NONCE_SIZE + USER_NONCE_SIZE + 32
 
+        ok = CT_sig_verify(resp["pubkey"], sha256s(msg), resp["sig"])
+        print("sig check ok?", ok)
```

Comment:
as mentioned we do not do this check in - added for verification

Reply:
even `master pubkey + master chain code` does NOT verify https://github.com/scgbckbone/coinkite-tap-proto/pull/1/commits/611a4ce30fc9f4f374ed82641752cae87ab2fa56

maybe same bug as with sim, where new nonce in not picked in the right time (i.e before creating msg digest)